lib/Conversion/MooreToCore/MooreToCore.cpp

```diff
@@ -155,6 +155,36 @@ struct ConstantOpConv : public OpConversionPattern<ConstantOp> {
   }
 };
 
+struct NamedConstantOpConv : public OpConversionPattern<NamedConstantOp> {
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(NamedConstantOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    Type resultType = typeConverter->convertType(op.getResult().getType());
+    std::string symStr;
+    switch (op.getKind()) {
+    case NamedConst::Parameter:
+      symStr = "parameter";
+      break;
+    case NamedConst::LocalParameter:
+      symStr = "localparameter";
+      break;
+    case NamedConst::SpecParameter:
+      symStr = "specparameter";
+      break;
+    }
+    symStr += ":";
+    symStr += op.getNameAttr().str();
+    auto symAttr = rewriter.getStringAttr(symStr);
```

Comment:
nit: Could there use Twine()(lightweight) and llvm::StringRef() to connect string literal? Like
` auto symAttr = rewriter.getStringAttr(symStr + Twine(":") + adaptor.getName());`

Reply:
Sorry I will be careful.